python/ray/util/sgd/torch/training_operator.py

```diff
@@ -33,57 +44,85 @@ def _is_multiple(component):
+
+    You must subclass this class and override the ``setup`` method to define
+    your training components such as the model, optimizer, data, loss,
+    and scheduler. When you pass this class to ``TorchTrainer``, a copy of
+    this class will be made on each worker.
+
+    .. code-block:: python
+
+        class MyTrainingOperator(TrainingOperator):
+
+            def setup(config):
+                model = nn.Linear(1, 1)
+                optimizer = torch.optim.SGD(
+                    model.parameters(), lr=config.get("lr", 1e-4))
+                loss = torch.nn.MSELoss()
+
+                batch_size = config["batch_size"]
+                train_data, val_data = LinearDataset(2, 5), LinearDataset(2, 5)
+                train_loader = DataLoader(train_data, batch_size=batch_size)
+                val_loader = DataLoader(val_data, batch_size=batch_size)
+
+                self.model, self.optimizer = self.register(
+                                                    models=model,
+                                                    optimizers=optimizer,
+                                                    train_loader=train_loader,
+                                                    validation_loader=val_loader,
+                                                    criterion=loss)
+        trainer = TorchTrainer(
+            training_operator_cls=MyTrainingOperator,
+            config={"batch_size": 32},
+            use_gpu=True
+        )
+        for i in range(4):
+            trainer.train()
+
+    This class provides default implementations for training and validation.
+    Make sure you set ``self.model``, ``self.optimizer``, and
+    ``self.criterion`` to leverage the default training and validation loops.
+    If ``self.scheduler`` is set, it will only be called at a batch or epoch
+    frequency, depending on the user parameter. Be sure to set
+    ``scheduler_step_freq`` in ``TorchTrainer`` to either "batch" or "epoch"
+    to increment the scheduler correctly during training. If using a
+    learning rate scheduler that depends on validation loss, you can use
+    ``trainer.update_scheduler``.
+
+    If you want to provide custom training and validation loops, you can do
+    so using this class as well. There are two granularities that
     you can provide customization: per epoch or per batch.
     You do not need to override both.
 
     .. image:: raysgd-custom.jpg
         :scale: 80%
         :align: center
 
+    If you are using multiple models, optimizers, or schedulers, you must
+    implement custom training and validation.
+
     Raises:
-        ValueError if multiple models/optimizers/schedulers are provided.
-            You are expected to subclass this class if you wish
-            to train over multiple models/optimizers/schedulers.
+        ValueError if custom training or validation is not provided,
+        and ``self.model``, ``self.optimizer``, or ``self.criterion``
+        are not set.
```

Comment:
I think this doesn't render properly on readthedocs?

Reply:
I think it needs the bottom row to be indented

Reply:
Fixed.